gapic-generator-cloud/templates/cloud/client/method/def/_client.erb

```diff
@@ -22,12 +22,26 @@
 #
 def <%= method.name %> requests, options: nil, &block
   unless requests.is_a? Enumerable
-    raise ArgumentError, "requests must be an Enumerable"
+    if requests.respond_to? :to_enum
+      requests = requests.to_enum
+    else
+      raise ArgumentError, "requests must be an Enumerable"
+    end
   end
 
   requests = requests.lazy.map do |request|
+    # Google::Gax::Protobuf.coerce request, to: <%= method.request_type %>
     Google::Gax.to_proto request, <%= method.request_type %>
   end
 
-  <%= method.ivar %>.call(requests, options, &block)
+  # Converts hash and nil to an options object
+  options = Google::Gax::ApiCall::Options.new options.to_h if options.respond_to? :to_h
+  header_params = {} # { name: request.name }
+  request_params_header = header_params.map { |k, v| "#{k}=#{v}" }.join("&")
+  metadata = @metadata.merge "x-goog-request-params".freeze => request_params_header
+  retry_policy = { retry_codes: [GRPC::Core::StatusCodes::UNAVAILABLE] }
```

Comment:
`UNAVAILABLE` cannot be a default.

Unfortunately, we will probably need a way to configure this in practice because it is correct in many cases. However, there is no configuration flag for it in the spec so we'll have to come up with one, likely via the yaml file you introduced. I'd suggest removing this as a default for now and we can decide how to allow a user to configure it in a later PR.

Reply:
This is only a placeholder to demonstrate where the default codes will go. The same for the header params a few lines above. This is only showing where the values will be generated once we can determine what the correct values are.

Reply:
Removed.